packages/flutter_tools/lib/src/base/io.dart

```diff
@@ -99,3 +101,76 @@ void setExitFunctionForTests([ExitFunction exitFunction]) {
 void restoreExitFunction() {
   _exitFunction = _defaultExitFunction;
 }
+
+/// A portable version of [io.ProcessSignal].
+///
+/// Listening on signals that don't exist on the current platform is just a
+/// no-op. This is in contrast to [io.ProcessSignal], where listening to
+/// non-existent signals throws an exception.
+class ProcessSignal implements io.ProcessSignal {
+  static const ProcessSignal SIGWINCH = const _PosixProcessSignal._(io.ProcessSignal.SIGWINCH);
+  static const ProcessSignal SIGTERM = const _PosixProcessSignal._(io.ProcessSignal.SIGTERM);
+  static const ProcessSignal SIGUSR1 = const _PosixProcessSignal._(io.ProcessSignal.SIGUSR1);
+  static const ProcessSignal SIGUSR2 = const _PosixProcessSignal._(io.ProcessSignal.SIGUSR2);
+
+  static final ProcessSignal SIGINT = SigintProcessSignal.instance; // ignore: non_constant_identifier_names
+
+  final io.ProcessSignal _wrappedSignal;
+
+  const ProcessSignal._(this._wrappedSignal);
+
+  @override
+  Stream<ProcessSignal> watch() {
+    return _wrappedSignal.watch().map((io.ProcessSignal signal) => this);
+  }
+
+  @override
+  String toString() => _wrappedSignal.toString();
+}
+
+/// A [ProcessSignal] that is only available on Posix platforms.
+///
+/// Listening to a [_PosixProcessSignal] is a no-op on Windows.
+class _PosixProcessSignal extends ProcessSignal {
+
+  const _PosixProcessSignal._(io.ProcessSignal wrappedSignal) : super._(wrappedSignal);
+
+  @override
+  Stream<ProcessSignal> watch() {
+    if (platform.isWindows)
+      return new Stream<ProcessSignal>.empty();
+    return super.watch();
+  }
+}
+
+// TODO(goderbauer): remove when https://github.com/dart-lang/sdk/issues/28995 is resolved.
+class SigintProcessSignal extends ProcessSignal {
+  static final SigintProcessSignal instance = new SigintProcessSignal._(io.ProcessSignal.SIGINT);
+  bool _isInitialized = false;
+
+  final StreamController<ProcessSignal> _controller = new StreamController<ProcessSignal>();
+
+  SigintProcessSignal._(io.ProcessSignal wrappedSignal) : super._(wrappedSignal);
+
+  @override
+  Stream<ProcessSignal> watch() {
+    init();
+    return _controller.stream;
+  }
+
+  void init() {
```

Comment:
That's exactly the reason why I am not doing the init stuff in the constructor. The init stuff needs to happen as early as possible for the work around to work.